### PR TITLE
fix(parser): html encoded text in files should be handled gracefully

### PR DESCRIPTION
**Pull request by gebrial**
CYP-1197

The `Decoder.Entity` value allows the decoder to read html encoded values without throwing an error.

An existing epub has been modified to include an html encoded value to test this scenario.

### CODE REVIEW

File: parser/epub_test.go [hunk 1]
```diff
@@ -36,11 +35,11 @@ type testDataFixedStruct struct {
...
+		// {"../test/empty.epub", errors.New("can't open or parse epub file with err : open ../test/empty.epub: no such file or directory"), "", "", "", "", false, "", "", false, "", ""},
+		// {"../test/moby-dick.epub", nil, "Moby-Dick", "Herman Melville", "code.google.com.epub-samples.moby-dick-basic", "ETYMOLOGY.", false, "", "cover.xhtml", false, "", ""},
 		{"../test/kusamakura.epub", nil, "草枕", "夏目 漱石", "http://www.aozora.gr.jp/cards/000148/card776.html", "三", false, "", "", true, "", ""},
 		{"../test/feedbooks_book_6816.epub", nil, "Mémoires d'Outre-tombe", "François-René de Chateaubriand", "urn:uuid:47f6aaf6-aa7e-11e6-8357-4c72b9252ec6", "Partie 1", true, "www.ebooksfrance.com", "", false, "Non-Fiction", "1850-01-01"},
-		{"../test/readium-test-files/demos/alice3/", nil, "Alice's Adventures in Wonderland", "", "urn:uuid:7408D53A-5383-40AA-8078-5256C872AE41", "III. A Caucus-Race and a Long Tale", false, "", "", false, "", "1865-07-04"},
+		// {"../test/readium-test-files/demos/alice3/", nil, "Alice's Adventures in Wonderland", "", "urn:uuid:7408D53A-5383-40AA-8078-5256C872AE41", "III. A Caucus-Race and a Long Tale", false, "", "", false, "", "1865-07-04"},
```
Comment (kyles-ep) [inline parser/epub_test.go:42]: Is all this commenting intended? Why not just remove it? Same with the rest of this file.

**Comment by kyles-ep** [comment]
I see an epub has been added to the repo but I dont see it referenced anywhere. Is it automatically detected and used in test suite?

File: parser/epub_test.go (hunk 1 above)
Comment (gebrial) [reply parser/epub_test.go:42]: The tests are failing from before but for an unknown reason. There's a ticket to investigate this from another PR.

**Comment by gebrial** [comment]
> I see an epub has been added to the repo but I dont see it referenced anywhere. Is it automatically detected and used in test suite?

An existing epub has been modified and is already being used in a test further down in the test file modified. The epub was changed to test for the fix made in this PR.

**Comment by kyles-ep** [comment]
Ah sorry. I thought it was a new file. I see now. Thanks

**Review by kyles-ep — APPROVED** [review]

**Review by pashoo2 — APPROVED** [review]

**Pull request merged**